src/MatrixTwitter.js

```diff
@@ -0,0 +1,987 @@
+ * timelines are no longer processed.
+ */
+MatrixTwitter.prototype.stop_timeline = function() {
+    if (this.timeline_intervalID) {
+        clearInterval(this.timeline_intervalID);
+        this.timeline_intervalID = null;
+    }
+}
+
+
+/**
+ * MatrixTwitter.prototype.start_timeline - Start the timeline timer so that
+ * timelines will be processed in turn.
+ */
+MatrixTwitter.prototype.start_timeline = function() {
+    this.timeline_intervalID = setInterval(() => {this._process_timeline();}, this.timeline_period);
+}
+
+/*
+  Hashtag functions
+*/
+
+
+/**
+ * MatrixTwitter.prototype.start_hashtag - Start the hashtag timer so that
+ * hashtags are processed in turn.
+ */
+MatrixTwitter.prototype.start_hashtag = function(){
+  this.hashtag_intervalID = setInterval(() => {this._process_hashtag_feed();}, this.hashtag_period);
+}
+
+/**
+ * MatrixTwitter.prototype.stop_hashtag - Stop the hashtag timer so that
+ * hashtags are no longer processed.
+ */
+MatrixTwitter.prototype.stop_hashtag = function(){
+    if (this.hashtag_intervalID) {
+        clearInterval(this.hashtag_intervalID);
+        this.hashtag_intervalID = null;
+    }
+}
+
+
+/**
+ * MatrixTwitter.prototype.add_hashtag_feed - Add a hashtag to be processed.
+ *
+ * @param  {string} hashtag The hasgtag to add (without the #).
+ * @param  {external:Entry} entry The entry for a room in the RoomBridgeStore.
+ */
+MatrixTwitter.prototype.add_hashtag_feed = function(hashtag,entry){
+    var obj = {
+        "hashtag": hashtag,
+        "entry": entry
+    };
+    this.hashtag_list.push(obj);
+    this.hashtag_queue.push(obj);
+    log.info('Twitter',"Added Hashtag Feed: %s",hashtag);
+}
+
+
+/**
+ * MatrixTwitter.prototype.remove_hashtag_feed - Remove a hashtag so that it
+ * will no longer be processed.
+ *
+ * @param  {string} hashtag The hasgtag to remove (without the #).
+ */
+MatrixTwitter.prototype.remove_hashtag_feed = function(hashtag){
+  const htfind = (feed) => { feed.hashtag == hashtag };
+  this.hashtag_list  = this.hashtag_list.splice( this.hashtag_list.findIndex(tlfind) ,1);
+  this.hashtag_queue = this.hashtag_queue.splice(this.hashtag_queue.findIndex(tlfind),1);
+}
+
+MatrixTwitter.prototype._process_hashtag_feed = function(){
+  if (this.hashtag_queue.length < 1) {
+      return;
+  }
+
+  var feed = this.hashtag_queue.shift();
+  var req = {
+    q: "%23"+feed.hashtag,
+    result_type: 'recent'
+  };
+  var since = feed.entry.remote.get("twitter_since");
+  if (since != undefined) {
+      req.since_id = since;
+  }
+
+  this.app_twitter.get('search/tweets', req, (error, results, response) => {
+      if(error){
+        log.error("Twitter","_process_timeline: GET /statuses/user_timeline returned: %s", error);
+        return;
+      }
+      if(results.statuses.length > 0){
+        feed.entry.remote.set("twitter_since", results.search_metadata.max_id_str);
+        this._bridge.getRoomStore().upsertEntry(feed.entry);
```

Comment:
`upsertEntry` returns a Promise, which you aren't doing anything with if it rejects.